dask/bytes/core.py

```diff
@@ -123,77 +121,97 @@ def read_bytes(urlpath, delimiter=None, not_zero=False, blocksize=2**27,
         raise IOError("%s resolved to no files" % urlpath)
     sizes = [fs.size(f) for f in names]
     out = []
+    # TODO: check fs for preferred locations of blocks here;
+    # and preferred blocksize?
+
     for size, name in zip(sizes, names):
-        blocksize = blocksize if blocksize is not None else size
-        offsets = list(range(0, size, blocksize))
+        bs = blocksize if blocksize is not None else size
+        offsets = list(range(0, size, bs))
         if len(offsets) > 1 and infer_compression(urlpath):
             raise ValueError('Cannot read compressed files (%s) in byte chunks,'
                              'use blocksize=None' % infer_compression(urlpath))
         if not_zero:
             offsets[0] = 1
         keys = ['read-block-%s-%s' % (offset, tokenize(name,
                 compression, offset, kwargs)) for offset in offsets]
-        # TODO: check fs for preferred locations of blocks here;
-        # and preferred blocksize?
-        func = lambda f, off: read_block(myopen(f, 'rb'), off, blocksize, delimiter)
+
+        def func(f, off):
+            with myopen(f, 'rb') as f:
+                return read_block(f, off, bs, delimiter)
         out.append([delayed(func)(name, off, dask_key_name=key)
                     for (off, key) in zip(offsets, keys)])
     if sample is not True:
         nbytes = sample
     else:
         nbytes = 10000
     if sample:
-        fs, myopen = make_myopen(urlpath, compression)
-        sample = read_block(myopen(names[0], 'rb'), 0, nbytes, delimiter)
+        myopen = MakeMyopen(urlpath, compression)
+        with myopen(names[0], 'rb') as f:
+            sample = read_block(f, 0, nbytes, delimiter)
     return sample, out
 
 
-def make_myopen(urlpath, compression=None, text=False, encoding='utf8',
-                errors=None, **kwargs):
-    """ Makes a lambda which opens a file in one of the registered backends.
+class MakeMyopen(object):
```

Comment:
Maybe remove "My" from the name here.  To me "my" means that it's one particular example of a thing used in testing.